### PR TITLE
[FLAX] glue training example refactor

### PR DESCRIPTION
**Pull request by kamalkraj**
# What does this PR do?

refactor glue training example similar to other flax training examples.

Fixes # (issue)

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/master/CONTRIBUTING.md#start-contributing-pull-requests),
      Pull Request section?
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/master/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/master/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

@patil-suraj @sgugger

### CODE REVIEW

**Review by sgugger — APPROVED** [review]
Thanks for refactoring this script!

**Review by patrickvonplaten — COMMENTED** [review]
Thanks for the refactor! @patil-suraj - we should add example tests for Flax as soon as possible so that we have an easier time reviewing such PRs.

Let's maybe do this before merging this PR.

**Comment by patrickvonplaten** [comment]
Ping here

**Comment by kamalkraj** [comment]
@patil-suraj @patrickvonplaten 
 rebased with current master

**Comment by patrickvonplaten** [comment]
@patil-suraj - this has been open since a while now, could you please take care of it?

File: examples/flax/text-classification/run_flax_glue.py [hunk 1]
```diff
@@ -82,7 +87,7 @@ def parse_args():
         "--validation_file", type=str, default=None, help="A csv or a json file containing the validation data."
     )
     parser.add_argument(
```
Comment (patil-suraj) [inline examples/flax/text-classification/run_flax_glue.py:89]: Could we also use the argument dataclasses here, to be consistent with other flax examples ?

File: examples/flax/text-classification/run_flax_glue.py [hunk 2]
```diff
@@ -493,58 +506,88 @@ def eval_step(state, batch):
     # make sure weights are replicated on each device
     state = replicate(state)
 
-    for epoch in range(1, num_epochs + 1):
-        logger.info(f"Epoch {epoch}")
-        logger.info("  Training...")
+    step_per_epoch = len(train_dataset) // train_batch_size
```
Comment (patil-suraj) [inline examples/flax/text-classification/run_flax_glue.py:509]:
```suggestion
    steps_per_epoch = len(train_dataset) // train_batch_size
```

File: examples/flax/text-classification/run_flax_glue.py [hunk 3]
```diff
@@ -493,58 +506,88 @@ def eval_step(state, batch):
...
-                repo.push_to_hub(commit_message=f"Saving weights and logs of epoch {epoch}", blocking=False)
+        for step, batch in enumerate(
+            tqdm(
+                glue_train_data_collator(input_rng, train_dataset, train_batch_size),
+                total=step_per_epoch,
+                desc="Training...",
+                position=1,
+            ),
+            1,
+        ):
```
Comment (patil-suraj) [inline examples/flax/text-classification/run_flax_glue.py:529]: (nit) This feels a bit hard to read. Could we maybe refactor it like this


```python
train_loader = glue_train_data_collator(input_rng, train_dataset, train_batch_size)
for step, batch in enumerate(train_loader, total=step_per_epoch, desc="Training...", position=1), 1):
    ....
```

File: examples/flax/text-classification/run_flax_glue.py [hunk 4]
```diff
@@ -493,58 +506,88 @@ def eval_step(state, batch):
...
+            if (cur_step % args.eval_steps == 0 or cur_step % step_per_epoch == 0) and cur_step > 0:
+
+                eval_metrics = {}
+                # evaluate
+                for batch in tqdm(
+                    glue_eval_data_collator(eval_dataset, eval_batch_size),
+                    total=len(eval_dataset) // eval_batch_size,
+                    desc="Evaluating ...",
+                    position=2,
+                ):
```
Comment (patil-suraj) [inline examples/flax/text-classification/run_flax_glue.py:557]: same comment as above

**Review by patil-suraj — APPROVED** [review]
Thanks a lot for the refactor @kamalkraj! Just left a couple of nits :)

File: examples/flax/text-classification/run_flax_glue.py (hunk 4 above)
Comment (kamalkraj) [reply examples/flax/text-classification/run_flax_glue.py]: done

File: examples/flax/text-classification/run_flax_glue.py (hunk 3 above)
Comment (kamalkraj) [reply examples/flax/text-classification/run_flax_glue.py]: done

File: examples/flax/text-classification/run_flax_glue.py (hunk 2 above)
Comment (kamalkraj) [reply examples/flax/text-classification/run_flax_glue.py]: done

File: examples/flax/text-classification/run_flax_glue.py (hunk 1 above)
Comment (kamalkraj) [reply examples/flax/text-classification/run_flax_glue.py]: done

**Comment by kamalkraj** [comment]
@patil-suraj refactored according to comments

**Review by patrickvonplaten — COMMENTED** [review]
Feel free to merge @patil-suraj, if you want :-)

**Pull request merged**